internal/backend/k8s.go

```diff
@@ -69,6 +77,28 @@ func NewK8sPersister(config *rest.Config, logger log.Logger) (*K8sPersister, err
 		return nil, fmt.Errorf("error creating cluster cache: %w", err)
 	}
 
+	cache.IndexField(context.Background(), &api.AccessRequest{}, usernameField, func(obj client.Object) []string {
+		ar, ok := obj.(*api.AccessRequest)
+		if !ok || ar == nil || ar.Spec.Subject.Username == "" {
```

Comment:
Are you suggesting to remove it to increase readability or let it there as a safety net?

Reply:
just a heads up = mainly for your awareness (in case you didn't know)

I implemented index creation in the controller code without it. So.. if we want to keep it consistent with the rest of the project, I would say lets remove it. Safety net is a valid argument as well. I am not strongly opinionated about this and there is no right/wrong decision to make. It is your call! :)